src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultWorkspaceChangedPublisher.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal interface IWorkspaceSemanticTokensRefreshPublisher
+    {
+        public void PublishWorkspaceSemanticTokensRefresh();
+    }
+
+    internal class DefaultWorkspaceSemanticTokensRefreshPublisher : IWorkspaceSemanticTokensRefreshPublisher
```

Comment:
Should make this IDisposable and dispose the work queue on dispose. That way if someone closes the solution we cleanup outstanding requests.

Reply:
Will also need to update the file name

Reply:
> Should make this IDisposable and dispose the work queue on dispose.

I'm surprised there's not an analyzer that yells at me about that, I kinda assumed there was and if I wasn't getting yelled at I was safe.